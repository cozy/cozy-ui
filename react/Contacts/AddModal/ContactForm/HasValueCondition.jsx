/*
  Wrapper for react-final-form field that renders it children only if the field
  with the given name has a truthy value or if the other condition is fulfilled
*/
import PropTypes from 'prop-types'
import React from 'react'
import { Field } from 'react-final-form'

const HasValueCondition = ({ children, otherCondition, name }) => {
  return (
    <Field name={name} subscription={{ value: true }}>
      {({ input: { value } }) =>
        (otherCondition !== undefined && otherCondition) || value
          ? children
          : null
      }
    </Field>
  )
}

HasValueCondition.defaultProps = {
  otherCondition: undefined
}

HasValueCondition.propTypes = {
  children: PropTypes.element.isRequired,
  name: PropTypes.string.isRequired,
  otherCondition: PropTypes.bool
}

export default HasValueCondition
