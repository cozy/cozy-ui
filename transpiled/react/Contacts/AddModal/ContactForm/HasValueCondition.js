/*
  Wrapper for react-final-form field that renders it children only if the field
  with the given name has a truthy value or if the other condition is fulfilled
*/
import PropTypes from 'prop-types';
import React from 'react';
import { Field } from 'react-final-form';

var HasValueCondition = function HasValueCondition(_ref) {
  var children = _ref.children,
      otherCondition = _ref.otherCondition,
      name = _ref.name;
  return /*#__PURE__*/React.createElement(Field, {
    name: name,
    subscription: {
      value: true
    }
  }, function (_ref2) {
    var value = _ref2.input.value;
    return otherCondition !== undefined && otherCondition || value ? children : null;
  });
};

HasValueCondition.defaultProps = {
  otherCondition: undefined
};
HasValueCondition.propTypes = {
  children: PropTypes.element.isRequired,
  name: PropTypes.string.isRequired,
  otherCondition: PropTypes.bool
};
export default HasValueCondition;