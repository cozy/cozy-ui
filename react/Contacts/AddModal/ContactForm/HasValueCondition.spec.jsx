import React from 'react'
import { Form } from 'react-final-form'
import renderer from 'react-test-renderer'

import HasValueCondition from './HasValueCondition'

describe('HasValueCondition component', () => {
  it('should render its children if value is truthy', () => {
    const jsx = (
      <Form
        initialValues={{ foo: 'anything' }}
        onSubmit={jest.fn()}
        render={({ handleSubmit }) => (
          <form onSubmit={handleSubmit}>
            <HasValueCondition name="foo">
              <span>Foo should be here</span>
            </HasValueCondition>
          </form>
        )}
      />
    )
    const tree = renderer.create(jsx).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should not render its children if value is falsy', () => {
    const jsx = (
      <Form
        initialValues={{ foo: '' }}
        onSubmit={jest.fn()}
        render={({ handleSubmit }) => (
          <form onSubmit={handleSubmit}>
            <HasValueCondition name="foo">
              <span>Foo should not be here</span>
            </HasValueCondition>
          </form>
        )}
      />
    )
    const tree = renderer.create(jsx).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render its children if value is falsy but other condition is fulfilled', () => {
    const jsx = (
      <Form
        initialValues={{ foo: '' }}
        onSubmit={jest.fn()}
        render={({ handleSubmit }) => (
          <form onSubmit={handleSubmit}>
            <HasValueCondition name="foo" otherCondition={true}>
              <span>Foo should be here</span>
            </HasValueCondition>
          </form>
        )}
      />
    )
    const tree = renderer.create(jsx).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should not render its children if value is falsy and other condition is not fulfilled', () => {
    const jsx = (
      <Form
        initialValues={{ foo: '' }}
        onSubmit={jest.fn()}
        render={({ handleSubmit }) => (
          <form onSubmit={handleSubmit}>
            <HasValueCondition name="foo" otherCondition={false}>
              <span>Foo should not be here</span>
            </HasValueCondition>
          </form>
        )}
      />
    )
    const tree = renderer.create(jsx).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
