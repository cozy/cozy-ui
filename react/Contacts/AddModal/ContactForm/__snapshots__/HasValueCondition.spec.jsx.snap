// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HasValueCondition component should not render its children if value is falsy 1`] = `
<form
  onSubmit={[Function]}
/>
`;

exports[`HasValueCondition component should not render its children if value is falsy and other condition is not fulfilled 1`] = `
<form
  onSubmit={[Function]}
/>
`;

exports[`HasValueCondition component should render its children if value is falsy but other condition is fulfilled 1`] = `
<form
  onSubmit={[Function]}
>
  <span>
    Foo should be here
  </span>
</form>
`;

exports[`HasValueCondition component should render its children if value is truthy 1`] = `
<form
  onSubmit={[Function]}
>
  <span>
    Foo should be here
  </span>
</form>
`;
